docs/content/preview/contribute/core-database/merge-with-upstream-repositories.md

```diff
@@ -0,0 +1,1017 @@
+   This is only permissible if the current upstream repository referenced commit is not a YugabyteDB commit (e.g. there is no `yb-pg11` branch since a `pgaudit/pgaudit` commit is used directly).
+   Then, the target version can directly be used.
+   (In this example, this shortcut can not be used.)
+1. Otherwise, follow the steps to identify commits to cherry-pick and create a `yb-pg15` branch as in [squash non-direct-descendant merge](#squash-non-direct-descendant-merge).
+1. On the [yugabyte/yugabyte-db][repo-yugabyte-db] repo, [subtree merge](#git-subtrees) this branch.
+
+## General advice
+
+### Git author information
+
+Each commit has the following six pieces of metadata.
+(See `git commit --help` "COMMIT INFORMATION" section for more details.)
+
+- GIT_AUTHOR_NAME
+- GIT_AUTHOR_EMAIL
+- GIT_AUTHOR_DATE
+- GIT_COMMITTER_NAME
+- GIT_COMMITTER_EMAIL
+- GIT_COMMITTER_DATE
+
+Normally, `git cherry-pick`ing a commit preserves that commit's author information.
+Otherwise, the author information is generated with your configuration.
+`git commit --amend --reset-author` also regenerates the author information with your configuration.
+Take care not to overwrite the author information when you shouldn't.
+
+For Phorge, the author information of the latest commit being landed is transferred to the merge or squash commit constructed by `arc land`.
+
+### Find PostgreSQL back-patch commits
+
+When PostgreSQL back-patches a commit, the commit message is generally unchanged.
+Therefore, one way to find all back-patches of a given commit is by filtering on the commit title: `git --grep '<title>' --all`.
+Do be aware to neutralize any regex special characters when using this command, for example by replacing them with `.` or `.*` or omitting them if they are at one of the ends.
+For example, title `Reject substituting extension schemas or owners matching ["$'\].` can be searched by `git --grep 'Reject substituting extension schemas or owners matching ' --all`.
+
+If you already have a certain branch in mind, it is more efficient to search in that branch directly.
+For example, if you are interested in finding whether a certain `master` commit has been back-patched to `REL_15_STABLE`, use `git --grep '<title>' REL_15_STABLE`.
+If you are interested in finding whether a certain `REL_15_STABLE` back-patch commit is present in `REL_18_1`, use `git --grep '<title>' REL_18_1`.
+
+### Testing PostgreSQL
+
+When doing a point-import or merge of upstream PostgreSQL, often that leads to creating a commit in [yugabyte/postgres][repo-postgres] repo.
+One should make sure that tests pass after that commit to catch logical merge conflicts.
+
+There are many tests scattered around different Makefiles, generally run using `make check`.
+Ideally, all should pass, but here are a few that one should pay special attention to since [yugabyte/yugabyte-db][repo-yugabyte-db] also partially uses them:
```

Comment:
```suggestion
Ideally, all should pass, but here are a few that you should pay special attention to since [yugabyte/yugabyte-db][repo-yugabyte-db] also partially uses them:
```